### ISSUE TITLE
Bug: text mode: rabbits walking/falling from the the world causes out of bounds exception

### ISSUE THREAD

**Issue by colonelfazackerley** [andybalaam/rabbit-escape#149]
Rabbits walking/falling from the the world causes out of bounds exception.

**Comment by andybalaam** [comment]
Yes, a bug

**Comment by colonelfazackerley** [comment]
I am working on this now

**Comment by colonelfazackerley** [comment]
This bug is back in master

**Comment by andybalaam** [comment]
Ouch. Did we fix it then lose the fix somehow?

**Comment by andybalaam** [comment]
I am working on a test for the command line ui but it is harder than I expected.

**Comment by andybalaam** [comment]
I am planning on using the Unix tool `expect`

**Comment by colonelfazackerley** [comment]
beat you
```
#!/usr/bin/expect
# sudo apt-get install expect

set timeout 20

spawn java -cp rabbit-escape-engine/bin/:rabbit-escape-render/bin/:rabbit-escape-ui-text/bin/:rabbit-escape-ui-swing/bin/ rabbitescape.ui.text.TextMain

# Start Game
send "1\r"

# Hard
send "3\r"

# Level 1
send "1\r"

# play
# :solution.2=bridge;(11,7);11;(12,7);2;(12,7);4;(12,7);2;(10,6);16;(13,4);5;(13,4);14

expect ">" 
send "bridge\r"
send "(11,7)\r"
send "11\r"
send "(12,7)\r"
send "2\r"
send "(12,7)\r"
send "4\r"
send "(12,7)\r"
send "2\r"
send "(10,6)\r"
send "16\r"
send "(13,4)\r"
send "5\r"
send "(13,4)\r"
send "14\r"


interact
```

**Comment by colonelfazackerley** [comment]
I think the bug was only partly fixed before. mostly it's OK for rabbits to 
- walk off the edge
- fall off the bottom
- climb a slope off the top

The bug seems to happen when they are building a bridge off the side or top. it's intermittent. I have not understood all the conditions for manifestation

**Comment by andybalaam** [comment]
I am working on a Makefile target `make slowtest` that will run system tests like this and possibly later the Android smoke test and similar. If you send me a pull request with the above script in a dir called slowtests I will gladly use it. Thanks!

**Comment by colonelfazackerley** [comment]
Here's a plan
- some code (java or bash, probably java) pulls the solution from a rel file and turns it into an expect script that plays the game in text mode.
- the expect script runs and the new solution is retrieved from that session.
- this solution may not be char-for-char identical to the one in the rel file because it may have been simplified. it should however do the same thing, and it should work.
- this solution is run using the existing solution running framework.

**Comment by andybalaam** [comment]
Sounds good for testing solution-related stuff. A more basic test we need is just that the command line ui works at all, in which we can hard code a solution.

**Comment by andybalaam** [comment]
Also I'd have no problem requiring that solutions in published levels are fully simplified, so we can check char for char.

**Comment by colonelfazackerley** [comment]
This bug was reopened for bridges being built out of bounds.

**Comment by colonelfazackerley** [comment]
Fixed for bridge building by c1aa840e6ef781a08091598d47e3659d2e0ab713
